### DISCUSSION TITLE
Access fitted models

### DISCUSSION THREAD

**Discussion by lorenzoFabbri** [tlverse/sl3#416] [Q&A] [answered]
In some cases I'd like to have access to the fitted models. For instance, to pass the model to other packages like `marginaleffects`. I see, for instance, that a `glm` is fitted [here](https://github.com/tlverse/sl3/blob/6544257626080f96fb37595a8a1f77a7c80918d1/R/Lrnr_glm.R#L89), but `fit_object` is a list of properties and I cannot seem to find the actual `glm` model. Am I missing something? Thanks!

**Comment by nhejazi** [comment, marked as answer]
thanks for the question, @lorenzoFabbri -- the short answer is that the fitted model object is unlikely to be preserved at the level that `marginaleffects` might require. you'll see just after the line that you point to that the S3 class that `glm` returns is stripped of much of its contents (https://github.com/tlverse/sl3/blob/devel/R/Lrnr_glm.R#L94-L104). this is done so as to minimize the memory footprint of the fitted model objects (of which there may be very many in a large stack)
